source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -816,10 +822,17 @@ void Filter::onMaxDownstreamConnectionDuration() {
 }
 
 void Filter::onAccessLogFlushInterval() {
-  for (const auto& access_log : config_->accessLogs()) {
-    access_log->log(nullptr, nullptr, nullptr, getStreamInfo());
+  if (getStreamInfo().streamState() &&
```

Comment:
I think  you can refactor this to be cleaner if you combine `getStreamInfo().streamState()` condition in both clauses., same for HCM logic.

Reply:
Done